core/src/worker/workflow/machines/upsert_search_attributes_state_machine.rs

```diff
@@ -238,12 +346,61 @@ mod tests {
         .try_into()
         .unwrap();
 
-        OnEventWrapper::on_event_mut(&mut sm, cmd_scheduled_sm_event)
-            .expect("CommandScheduled should transition Created -> CommandIssued");
+        OnEventWrapper::on_event_mut(
+            &mut sm,
+            CommandType::UpsertWorkflowSearchAttributes
+                .try_into()
+                .unwrap(),
+        )
+        .expect("CommandScheduled should transition Created -> CommandIssued");
         assert_eq!(CommandIssued {}.to_string(), sm.state().to_string());
 
-        OnEventWrapper::on_event_mut(&mut sm, cmd_recorded_sm_event)
-            .expect("CommandRecorded should transition CommandIssued -> Done");
-        assert_eq!(Done {}.to_string(), sm.state().to_string());
+        let recorded_res = OnEventWrapper::on_event_mut(&mut sm, cmd_recorded_sm_event);
+        if nondetermistic {
+            assert_matches!(
+                recorded_res.unwrap_err(),
+                MachineError::Underlying(WFMachinesError::Nondeterminism(_))
+            );
+        } else {
+            recorded_res.expect("CommandRecorded should transition CommandIssued -> Done");
+            assert_eq!(Done {}.to_string(), sm.state().to_string());
+        }
+    }
+
+    #[tokio::test]
+    async fn upserting_change_version_directly_does_nothing() {
```

Comment:
Arguably it should only do nothing if the server supports SDK metadata and Core can upsert the search attribute.

Reply:
Done